src/experimental/tools/mediaCapabilitiesProber/probers/mediaContentType/index.ts

```diff
@@ -14,52 +14,44 @@
  * limitations under the License.
  */
 
-import { IMediaConfiguration } from "../../types";
+import { MediaSource_ } from "../../../../../compat";
+import {
+  IMediaConfiguration,
+  ProberStatus,
+} from "../../types";
 
 /**
+ * @param {Object} config
  * @returns {Promise}
  */
-function isContentTypeAPISupported(): Promise<void> {
-  return new Promise((resolve) => {
-    if (!("MediaSource" in window)) {
+export default function probeContentType(
+  config: IMediaConfiguration
+): Promise<[ProberStatus]> {
+  return new Promise(() => {
+    if (MediaSource_ == null) {
       throw new Error("MediaCapabilitiesProber >>> API_CALL: " +
         "MediaSource API not available");
     }
-    if (!("isTypeSupported" in (window as any).MediaSource)) {
+    if (typeof MediaSource_.isTypeSupported !== "string") {
       throw new Error("MediaCapabilitiesProber >>> API_CALL: " +
         "isTypeSupported not available");
     }
-    resolve();
-  });
-}
-
-/**
- * @param {Object} config
- * @returns {Promise}
- */
-export default function probeContentType(config: IMediaConfiguration): Promise<[number]> {
-  return isContentTypeAPISupported().then(() => {
     const contentTypes: string[] = [];
-    if (
-      config.video &&
-      config.video.contentType
-    ) {
+    if (config.video && config.video.contentType) {
       contentTypes.push(config.video.contentType);
     }
-    if (
-      config.audio &&
-      config.audio.contentType
-    ) {
+    if (config.audio && config.audio.contentType) {
       contentTypes.push(config.audio.contentType);
     }
-      if (contentTypes === null || !contentTypes.length) {
-        throw new Error("MediaCapabilitiesProber >>> API_CALL: " +
-          "Not enough arguments for calling isTypeSupported.");
+    if (contentTypes.length === 0) {
+      throw new Error("MediaCapabilitiesProber >>> API_CALL: " +
+        "Not enough arguments for calling isTypeSupported.");
+    }
+    for (let i = 0; i < contentTypes.length; i++) {
+      if (!MediaSource_.isTypeSupported(contentTypes[i])) {
+        return [ProberStatus.NotSupported];
       }
-      const result: [number] = [contentTypes.reduce((acc, val) => {
-        const support = (window as any).MediaSource.isTypeSupported(val) ? 2 : 0;
-        return Math.min(acc, support);
```

Comment:
2Smart4Me

Also you continue to iterate even when a NotSupported case is reached.

Much more readable by doing a good ol' for loop.